web/src/rtl.ts

```diff
@@ -139,9 +144,13 @@ export function get_direction(str) {
     return "ltr";
 }
 
-export function set_rtl_class_for_textarea($textarea) {
+export function set_rtl_class_for_textarea($textarea: JQuery<HTMLTextAreaElement>): void {
     // Set the rtl class if the text has an rtl direction, remove it otherwise
     let text = $textarea.val();
+    if (typeof text !== "string") {
+        /* istanbul ignore throw */
+        throw new TypeError("Expected $textarea to be a textarea.");
+    }
```

Comment:
There’s no such thing as `istanbul ignore throw`, only `if`, `else`, and `next`. Generally you want to ignore an entire branch with `if` or `else` ([documentation](https://github.com/gotwarlost/istanbul/blob/master/ignoring-code-for-coverage.md#the-interface)).

We could use a type-narrowing assertion helper like [minimalistic-assert](https://www.npmjs.com/package/minimalistic-assert) to avoid this repetitive coverage skipping pattern ([discussion](https://chat.zulip.org/#narrow/stream/6-frontend/topic/coverage.20for.20.60throw.20new.20TypeError.60/near/1542686)).

Reply:
That plan sounds good to me.